nsolve:added printer to mpmathprinter for incomple gamma functions

priyank-p0 opened the pull request:
previously the mpmathprinter didn't had any printer for upper gamma and lower gamma function. The two added printer gives the output in string format as required by mpmath gammainc.
The changes passed all the tests
Fixes #12173 
<!-- If this pull request fixes an issue please indicate which issue by typing "Fixes #NNNN" below.>

priyank-p0 committed:
nsolve:added printer to mpmathprinter for incomple gamma functions

previously the mpmathprinter didn't had any printer for upper gamma and lower gamma function. The two added printer gives the output in string format as required by mpmath gammainc.
The changes passed all the tests

pspbot7 commented:
Closes #12173

pspbot7 commented:
@asmeurer can you look at this?

pspbot7 commented:
@asmeurer can you merge this request, if it has no issue.

jksuom commented:
Please add a unit test.

pspbot7 commented:
@jksuom i have done the tests locally and my branch passes the tests, also can you tell me how to add a unit test since i am new to it.
thanks

jksuom commented:
A suitable `def test_issue_xxxxx()` can be added to sympy/utilities/tests/test_lambdify.py.

pspbot7 commented:
@jksuom i am not able to write a test, can you help me out

jksuom commented:
This PR should fix the issue reported on the mailing list: https://groups.google.com/forum/#!topic/sympy/WVkLUGrpu8U. You could start by verifying that the given test file will pass locally on your machine. The file is probably too big to be incorporated in the Sympy test file, but you should extract a suitable part to show that lambdify will work as expected.

grassknoted commented:
The PR doesn't fix the issue reported on the mailing list. (https://groups.google.com/forum/#!topic/sympy/WVkLUGrpu8U)
The error message is as follows, if someone could help me a little, I would like to fix the issue and add unit  tests.
`Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/grassknoted/Desktop/sympy/sympy/utilities/decorator.py", line 91, in func_wrapper
    return func(*args, **kwargs)
  File "/home/grassknoted/Desktop/sympy/sympy/solvers/solvers.py", line 2822, in nsolve
    x = findroot(f, x0, J=J, **kwargs)
  File "/home/grassknoted/anaconda3/lib/python3.6/site-packages/mpmath/calculus/optimization.py", line 928, in findroot
    fx = f(*x0)
  File "<string>", line 1, in <lambda>
NameError: name 'uppergamma' is not defined`

priyank-p0 committed:
nsolve:added printer to mpmathprinter for incomple gamma functions

previously the mpmathprinter didn't had any printer for upper gamma and lower gamma function. The two added printer gives the output in string format as required by mpmath gammainc.
The changes passed all the tests

new changes: added the test cases for the mentioned solution

pspbot7 commented:
@asmeurer i have added the test cases, can you look at it?
